File: 367-valid-perfect-square/valid-perfect-square.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        long low = 1 ;
        long high = num;

        while(low <= high){
            long mid = low + (high - low)/2;
            if(mid*mid  == num){
                return true;
            }
            if(mid*mid > num){
                high = mid - 1;
            }
            else{
                low = mid + 1;
            }
        }
        return false;
    }
};

File: 367-valid-perfect-square/valid-perfect-square.java
class Solution {
    public boolean isPerfectSquare(int n) {

        long low = 1;
        long high = n;

        while (low <= high) {
            long mid = low + (high - low) / 2;

            long prod = mid * mid ; 

            if (prod == n) {
                return true;
            }

            else if (prod > n) {
                high = mid - 1;
            }

            else
                low = mid + 1;

        }
        return false;

    }
}